contracts/staking/SimpleStaking.sol

```diff
@@ -120,26 +120,28 @@ contract SimpleStaking is DSMath, Pausable, AbstractGoodStaking, StakingToken{
         if (tokenActual < tokenWithdraw) {
             tokenWithdraw = tokenActual;
         }
-        uint256 gdInterest =  InterestDistribution.withdrawStakeAndInterest(interestData, msg.sender, _amount);
+        //uint256 gdInterest =  InterestDistribution.withdrawStakeAndInterest(interestData, msg.sender, _amount);
         //Since we use generic ERC20 function we can just use its interface
-        ERC20 goodDollar = ERC20(address(avatar.nativeToken()));
-        fm.mintReward(msg.sender); // send rewards to user
+        //ERC20 goodDollar = ERC20(address(avatar.nativeToken()));
+        fm.mintReward(address(iToken), msg.sender); // send rewards to user
         _burn(msg.sender, _amount); // burn their staking tokens
         _decreaseProductivity(msg.sender, _amount);
-        require(goodDollar.transfer(msg.sender, gdInterest), "withdraw interest transfer failed");
+        //require(goodDollar.transfer(msg.sender, gdInterest), "withdraw interest transfer failed");
         require(token.transfer(msg.sender, tokenWithdraw), "withdraw transfer failed");
         emit StakeWithdraw(msg.sender, address(token), tokenWithdraw, token.balanceOf(address(this)));
     }
 
     function withdrawGDInterest() public {
         FundManager fm = FundManager(nameService.getAddress("FUND_MANAGER"));
         fm.transferInterest(address(this));
-        uint256 gdInterest = InterestDistribution.withdrawGDInterest(interestData, msg.sender);
-        ERC20 goodDollar = ERC20(address(avatar.nativeToken()));
-        require(goodDollar.transfer(msg.sender, gdInterest), "withdraw interest transfer failed");
+        //uint256 gdInterest = InterestDistribution.withdrawGDInterest(interestData, msg.sender);
```

Comment:
this is about the user withdrawing only their earned rewards not fundmanager collecting the iToken interest

Reply:
I'm deleting stuff that I commented out from old implementation @sirpy